lib/spack/env/cc

```diff
@@ -616,6 +616,9 @@ if [[ $SPACK_TEST_COMMAND == dump-args ]]; then
     IFS="
 " && echo "${full_command[*]}"
     exit
+elif [[ $SPACK_TEST_COMMAND =~ dump-env-* ]]; then
+    var=${SPACK_TEST_COMMAND#dump-env-}
+    eval echo "$0: \$var=\${$var}"
```

Comment:
```suggestion
    echo "$0: $var: ${!var}
```
you can use this bash trick to avoid eval

Reply:
Thanks, neat bash trick!

I committed your suggetion and updated the commit message, no other change in the new pushes.